### PR TITLE
OOP school library: set up associations

### PR DESCRIPTION
**Pull request by shafiuyushawu**
### Hello Reviewer 👍 
This is a pull request to create `OOP School Liabrary` setting up these classes `classroom`, `rental`,  `book`, and created relationships among them respetively.

# **Functionality implemented ✍️**

## **Classroom and Student Relationship**
- Created class `Classroom` with an `@label` instance variable
- Implemented `getter` and `setter` for `@label` using `attr_accessor`.
- Created relationship among classroom and Student
- Implemented functionalities for `getter` and `setter`.

## **Book and Rental Relationship**
- Created `Book` class with `@title` and `@author` as instance variables.
- Implemented getters and setters for `@title` and `@author` using `attr_accessor`.
- Created many-to-many relationship among `Student` and `Rental`.
- Established the belong-to-relationship between `Rental` and `Book`.
- Modified the constructor of  `Rental` to set `Book` and `Person` instances.

## **Person and Rental Relationship**
- Created a has-many relationship between `Person` and `Rental`.
- Established the belongs-to relationship between `Rental` and `Person`.

These are the changes for managing classrooms, students, books, and rentals with appropriate associations between the classes.

### CODE REVIEW

File: rental.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class Rental
+  attr_accessor :date, :book, :person
+
+  def initialize(book, person, date)
+    @date = date
+    @book = book
+    @person = person
+  end
```
Comment (TedLivist) [inline rental.rb]: - [x] When a rental is created it should add itself to the `rentals` of the book and person objects. That way the relationship between the book and the person through the rental is established.

  ```
  def initialize(book, person, date)
    @date = date
    @book = book
    @person = person

    book.rentals << self
    person.rentals << self
  end
  ```

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @shafiuyushawu,

Good job so far!
✔️ Establishment of association between classroom and student
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by julie-ify — APPROVED** [review]
Hi @shafiuyushawu,

## Highlights!
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: rental.rb (hunk 1 above)
Comment (shafiuyushawu) [reply rental.rb]: Resolved

**Comment by shafiuyushawu** [comment]
Hello @julie-ify ,
Thanks for the approval! much appreciation 👍

**Pull request merged**